plugin/storage/es/spanstore/reader.go

```diff
@@ -139,24 +140,43 @@ type timeRangeIndexFn func(indexName string, indexDateLayout string, startTime t
 
 type sourceFn func(query elastic.Query, nextTime uint64) *elastic.SearchSource
 
-func getTimeRangeIndexFn(archive, useReadWriteAliases bool) timeRangeIndexFn {
+func getTimeRangeIndexFn(archive, useReadWriteAliases bool, remoteReadClusters []string) timeRangeIndexFn {
 	if archive {
 		var archiveSuffix string
 		if useReadWriteAliases {
 			archiveSuffix = archiveReadIndexSuffix
 		} else {
 			archiveSuffix = archiveIndexSuffix
 		}
-		return func(indexName, indexDateLayout string, startTime time.Time, endTime time.Time) []string {
+		return addRemoteReadClusters(func(indexName, indexDateLayout string, startTime time.Time, endTime time.Time) []string {
 			return []string{archiveIndex(indexName, archiveSuffix)}
-		}
+		}, remoteReadClusters)
 	}
 	if useReadWriteAliases {
-		return func(indices string, indexDateLayout string, startTime time.Time, endTime time.Time) []string {
+		return addRemoteReadClusters(func(indices string, indexDateLayout string, startTime time.Time, endTime time.Time) []string {
 			return []string{indices + "read"}
+		}, remoteReadClusters)
+	}
+	return addRemoteReadClusters(timeRangeIndices, remoteReadClusters)
+}
+
+// Elasticsearch cross cluster query api example: GET /twitter,cluster_one:twitter,cluster_two:twitter/_search
+// Add a remote cluster prefix for each cluster and for each index and add it to the list of original indices
```

Comment:
nit: Please end comment sentences with a full-stop. 

Same as [above](https://github.com/jaegertracing/jaeger/pull/2874/files#diff-de5aae4da3cb03d3a73919aca1f6f38181b906dfd205bb0def4bf7b9d6042992R254) though I understand this convention is not followed in the adjacent functions below it.